debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorTask.java

```diff
@@ -159,13 +161,30 @@ public void commit() throws InterruptedException {
         }
     }
 
+    @Override
+    public void commitRecord(SourceRecord record) throws InterruptedException {
+        if (running.get()) {
+            Long recordLsn = (Long) record.sourceOffset().get(SourceInfo.LSN_KEY);
+
+            // If this record has an lsn and it's larger than the last seen lsn, then we've fully processed
+            // the events with the last seen lsn
+            if (recordLsn != null && recordLsn > previousLsn) {
+                // Update the last completely processed lsn to the previous batch's lsn
+                lastCompletelyProcessedLsn = previousLsn;
+
+                // Update previous lsn for comparison when processing the next batch
+                previousLsn = recordLsn;
+            }
```

Comment:
Can we add a log line of some kind to see if we ever see a previousLsn > recordLsn?